src/components/common/YourCompany/CompanyName/CorrectNameRequest.vue

```diff
@@ -138,36 +145,42 @@ export default class CorrectNameRequest extends Mixins(CommonMixin, NameRequestM
     this.$refs.correctNrForm.resetValidation()
   }
 
+  @Watch('validate')
+  private onValidate (): void {
+    this.$refs.correctNrForm.validate()
+  }
+
   /** Watch for form submission and emit results. */
   @Watch('formType')
   private async onSubmit (): Promise<any> {
     if (this.formType === CorrectionTypes.CORRECT_NEW_NR) {
       try {
         // Validate and return the name request data
-        const response: NrResponseIF = await this.validateNameRequest(
+        const nr: NrResponseIF = await this.validateNameRequest(
           this.nameRequestNumber,
           this.applicantPhone,
           this.applicantEmail
         )
 
-        // Parse the name request data
-        const nrCorrection: NrCorrectionIF = {
-          legalType: response.entity_type_cd,
-          nrNumber: this.nameRequestNumber,
-          legalName: this.getNrApprovedName(response) || '',
-          expiry: response.expirationDate,
-          status: response.state,
-          requestType: response.request_action_cd,
-          applicant: {
-            fullName: this.formatFullName(response.applicants),
-            fullAddress: this.formatFullAddress(response.applicants),
-            phoneNumber: response.applicants.phoneNumber,
-            emailAddress: response.applicants.emailAddress
-          }
+        if (this.getEntityType !== nr.legalType) {
+          // Invalid NR type, inform parent the process is done and prompt confirm dialog
+          this.emitDone()
+
+          const dialogContent = `<p class="info-text">This ${this.getCorpTypeDescription(nr.entity_type_cd)} ` +
+            `Name Request does not match the current business type ` +
+            `<b>${this.getCorpTypeDescription(this.getEntityType)}</b>.\n\n` +
+            `The Name Request type must match the business type before you can continue.</p>`
+
+          this.$root.$emit(
```

Comment:
Since this call is not async (we don't wait for it), what is the next thing the user sees, and what are their options?

Reply:
There only option here is 'OK' 
So no options really 😆